docs/USER_EXPERIENCE_DESIGN.md

```diff
@@ -220,6 +221,13 @@ primary sink rejects events.
 Users may need to intervene to update invalid configuration, for example in the
 case of rotating invalid API keys.
 
+### Sink log encoding
+
+When encoding logs for sinks, fields indicated by `log_schema` keys should be mapped to where the
+destination expects to find them in the encoded payloads. For example, if the sink requires
+a "hostname", it should fetch it from the log event using `log_schema.host_key` and place it in the
```

Comment:
Actually it should use these helpers https://github.com/vectordotdev/vector/blob/7570bb31e2f471e3ff8bc818c24e9bde3090818c/lib/vector-core/src/event/log_event.rs#L433